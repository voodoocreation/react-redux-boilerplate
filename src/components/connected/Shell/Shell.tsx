import cn from "classnames";
import Head from "next/head";
import * as React from "react";
import { InjectedIntl, injectIntl } from "react-intl";
import { connect } from "react-redux";

import { isServer } from "../../../helpers/dom";
import * as selectors from "../../../selectors/root.selectors";

interface IStoreProps {
  isLoading: boolean;
}

interface IProps extends IStoreProps {
  className?: string;
  intl: InjectedIntl;
}

class Shell extends React.Component<IProps> {
  public componentWillMount() {
    if (!isServer()) {
      const html = document.documentElement as HTMLHtmlElement;
      html.classList.add("isClientRendered");
    }
  }

  public render() {
    const { children, className, isLoading } = this.props;
    const { formatMessage } = this.props.intl;

    return (
      <article className={cn("Page", { isLoading }, className)}>
        <Head>
          <meta
            property="og:site_name"
            content={formatMessage({ id: "BRAND_NAME" })}
          />
        </Head>

        <main className="Page-body container" role="main">
          {children}
        </main>
      </article>
    );
  }
}

const mapStateToProps = (state: any) => ({
  isLoading: selectors.getPageIsLoading(state)
});

export default injectIntl<any>(connect<IStoreProps>(mapStateToProps)(Shell));
