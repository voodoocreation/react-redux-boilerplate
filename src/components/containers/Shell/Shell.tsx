import cn from "classnames";
import Head from "next/head";
import * as React from "react";
import { InjectedIntl, injectIntl } from "react-intl";

interface IProps {
  className?: string;
  intl: InjectedIntl;
}

class Shell extends React.Component<IProps> {
  public componentWillMount() {
    if (
      typeof document !== "undefined" &&
      !document.documentElement.classList.contains("isClientRendered")
    ) {
      document.documentElement.classList.add("isClientRendered");
    }
  }

  public render() {
    const { children, className } = this.props;
    const { formatMessage } = this.props.intl;

    return (
      <article className={cn("Page", className)}>
        <Head>
          <meta
            property="og:site_name"
            content={formatMessage({ id: "BRAND_NAME" })}
          />
        </Head>

        <main className="Page-body container" role="main">
          {children}
        </main>
      </article>
    );
  }
}

export default injectIntl<any>(Shell);
