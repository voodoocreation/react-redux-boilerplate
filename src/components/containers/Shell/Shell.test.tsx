import { render } from "enzyme";
import * as React from "react";

import Shell from "./Shell";

const setup = (fn: any) => {
  const props = {};

  return {
    actual: fn(<Shell {...props} />),
    props
  };
};

describe("<Shell />", () => {
  it("renders correctly", () => {
    const { actual } = setup(render);
    expect(actual).toMatchSnapshot();
  });
});
