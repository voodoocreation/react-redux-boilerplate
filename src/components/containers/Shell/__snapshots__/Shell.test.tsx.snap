// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Shell /> renders correctly 1`] = `
<article
  class="Page"
>
  <main
    class="Page-body container"
    role="main"
  />
</article>
`;
